### ISSUE TITLE
explicit_counter_loop false positive

### ISSUE THREAD

**Issue by asomers** [rust-lang-nursery/rust-clippy#3308]
The `explicit_counter_loop` lint wrongly detects that I'm using a variable as a loop counter.  The code snippet is below.
```rust
        let mut skips = 0;
        for i in 0..k {
            while erasures.contains(i + skips) {
                skips += 1;
            }
            let row = i + skips;
            for j in 0..k {
                dec_matrix_inv[k * i + j] =
                    self.enc_matrix[k * row + j];
            }
        }
```
And Clippy suggests "fixing" it in a way that will clearly change its function:
```
warning: the variable `skips` is used as a loop counter. Consider using `for (skips, i
tem) in 0..k.enumerate()` or similar iterators
   --> src/common/raid.rs:216:18
    |
216 |         for i in 0..k {
    |                  ^^^^
    |
    = note: #[warn(clippy::explicit_counter_loop)] on by default
    = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.212/index.html#explicit_counter_loop
```
> cargo +nightly clippy -V
clippy 0.0.212 (32b1d1fc 2018-10-05)
```

**Comment by JoshMcguigan** [comment]
Thanks for the report. I believe PR #3310 resolves this issue.
